### PR TITLE
Remove duplicate code to make contract fit

### PR DESCRIPTION
**Pull request by iainnash**

### CODE REVIEW

**Review by kulkarohan — APPROVED** [review]

File: foundry.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [profile.default]
 solc_version = '0.8.17'
 optimizer = true
-optimizer_runs = 300
+optimizer_runs = 500
```
Comment (oveddan) [inline foundry.toml:4]: nice

File: package.json [hunk 2]
```diff
@@ -51,7 +51,11 @@
   },
   "dependencies": {
     "@dotenv/cli": "^2.2.2",
+<<<<<<< HEAD
+    "@zoralabs/protocol-rewards": "^1.0.1",
+=======
```
Comment (oveddan) [inline package.json:56]: broken rebase

File: src/ERC721Drop.sol [hunk 3]
```diff
@@ -732,21 +756,29 @@ contract ERC721Drop is
...
-        _handleRewards(msg.value, quantity, pricePerToken, config.fundsRecipient != address(0) ? config.fundsRecipient : address(this), createReferral, mintReferral);
-
-        _mintNFTs(msgSender, quantity);
-
-        uint256 firstMintedTokenId = _lastMintedTokenId() - quantity;
-
-        _emitSaleEvents(msgSender, msgSender, quantity, pricePerToken, firstMintedTokenId, comment);
-
-        return firstMintedTokenId;
+<<<<<<< HEAD
```
Comment (oveddan) [inline src/ERC721Drop.sol:759]: broken rebase :(

File: src/ERC721Drop.sol [hunk 4]
```diff
@@ -452,7 +452,7 @@ contract ERC721Drop is
         onlyPublicSaleActive
         returns (uint256)
     {
-        return _handlePurchase(msg.sender, quantity, "");
```
Comment (oveddan) [inline src/ERC721Drop.sol:455]: can we remove this _handlePurchase function?

File: src/ERC721Drop.sol (hunk 3 above)
Comment (iainnash) [reply src/ERC721Drop.sol:759]: was trying to get this diff better, didn't work, ignore for now

**Comment by iainnash** [comment]
Moving to a new diff.

**Pull request closed without merging**